core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DiskBasedDMSchemaStorageProvider.java

```diff
@@ -129,9 +130,9 @@ public DiskBasedDMSchemaStorageProvider(String storePath) {
     return dataMapSchemas;
   }
 
-  @Override public void dropSchema(String dataMapName) throws IOException {
-    String schemaPath =
-        storePath + CarbonCommonConstants.FILE_SEPARATOR + dataMapName + ".dmschema";
+  @Override public void dropSchema(String dataMapName,String tableName) throws IOException {
+    String schemaPath = storePath + CarbonCommonConstants.FILE_SEPARATOR + tableName
+        + CarbonCommonConstants.UNDERSCORE + dataMapName + ".dmschema";
```

Comment:
I think it is better to record the datamap short name in the path also, so that we can know what kind of datamap it is by looking at the file name